### PR TITLE
quant_state of Params4bit became None after "to" method

### PR DESCRIPTION
**Pull request by sxndqc**
# What does this PR do?

The **kwargs was missing in line 401; and previously the new class of bitsandbytes Params4bits was not included in line 389, which leads to the missing of quant_state after moving to another device because the re-initialization in line 401 will lead to a default quant_state=None due to lack of **kwargs. The processing of Params4bit is now moved to line 399, but I also added **kwargs to line 401 in case.

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sxndqc** [comment]
fix bug for Params4bit

**Review by muellerzr — COMMENTED** [review]
Cc @younesbelkada

**Review by younesbelkada — COMMENTED** [review]
Hi @sxndqc 
Thanks ! In principle looks good, I have the same comment as: https://github.com/huggingface/accelerate/pull/2376#pullrequestreview-1843439784 could you run the slow tests and confirm it passes?

**Comment by sxndqc** [comment]
Hi @younesbelkada 
It seems that the test file for utils/modeling.py is test_modeling_utils
Here is the result:

============================= test session starts ==============================
platform darwin -- Python 3.11.7, pytest-8.0.0, pluggy-1.4.0
rootdir: /Users/shb/Downloads/accelerate-main
collected 23 items                                                             

tests/test_modeling_utils.py .....s.s..s.sss..s..s.s                     [100%]

======================== 14 passed, 9 skipped in 5.56s =========================

**Comment by sxndqc** [comment]
@younesbelkada I've tried this line before, but I got this message:

============================ no tests ran in 0.00s =============================
ERROR: file or directory not found: tests/quantization/bnb/test_4bit.py

I tried this on the downloaded zip from github.

**Comment by younesbelkada** [comment]
@sxndqc hmm strange, do you have this file on the cloned repo? https://github.com/huggingface/transformers/blob/main/tests/quantization/bnb/test_4bit.py

**Comment by sxndqc** [comment]
@younesbelkada Hmm, it seems this file is in transformers' repo but not in accelerate's repo. I cloned accelerate's repo and didn't find the test_4bit.py in its test/ subrepo. I also looked up the file in site-packages and neither transformers nor accelerate has a test/ repo.

**Comment by sxndqc** [comment]
@younesbelkada Hi Younes, would it be convenient for you to run the code with modified modeling.py on your end? My local machine is Mac and it seems the bnb wouldn't work here and I don't have that test file either my installed version or from the github repository. Thank you!

File: src/accelerate/utils/modeling.py [hunk 1]
```diff
@@ -386,7 +386,7 @@ def set_module_tensor_to_device(
         elif value is not None or not check_device_same(torch.device(device), module._parameters[tensor_name].device):
             param_cls = type(module._parameters[tensor_name])
             kwargs = module._parameters[tensor_name].__dict__
-            if param_cls.__name__ in ["Int8Params", "FP4Params"]:
+            if param_cls.__name__ in ["Int8Params", "FP4Params", "Params4bit"]:
```
Comment (BenjaminBossan) [inline src/accelerate/utils/modeling.py:389]: Given that we use `["Int8Params", "FP4Params", "Params4bit"]` already in [line 363](https://github.com/huggingface/accelerate/pull/2381/files#diff-d9b8e1f4731ccbfb7b66e8685b7dd7c500c1ae5bba510b610714e72a39049de0R363), how about putting this into a constant?

**Pull request closed without merging**